src/test/fuzz/descriptor_parse.cpp

```diff
@@ -3,11 +3,109 @@
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
 #include <chainparams.h>
+#include <key_io.h>
 #include <pubkey.h>
 #include <script/descriptor.h>
 #include <test/fuzz/fuzz.h>
 #include <util/chaintype.h>
 
+//! Types are raw compressed pubkeys, raw xonly pubkeys, raw privkeys (WIF), xpubs, xprvs.
+static constexpr uint8_t KEY_TYPES_COUNT{5};
+//! How many keys we'll generate in total.
+static constexpr uint8_t TOTAL_KEYS_GENERATED{std::numeric_limits<uint8_t>::max()};
+
+/**
+ * Converts a mocked descriptor string to a valid one. Every key in a mocked descriptor key is
+ * represented by 2 hex characters preceded by the '%' character. We parse the two hex characters
+ * as an index in a list of pre-generated keys. This list contains keys of the various types
+ * accepted in descriptor keys expressions.
+ */
+class MockedDescriptorConverter {
+    //! 256 keys of various types.
+    std::array<std::string, TOTAL_KEYS_GENERATED> keys_str;
+
+public:
+    // We derive the type of key to generate from the 1-byte id parsed from hex.
+    bool IdIsCPubKey(uint8_t idx) const { return idx % KEY_TYPES_COUNT == 0; }
+    bool IdIsXOnlyPubKey(uint8_t idx) const { return idx % KEY_TYPES_COUNT == 1; }
+    bool IdIsConstPrivKey(uint8_t idx) const { return idx % KEY_TYPES_COUNT == 2; }
+    bool IdIsXpub(uint8_t idx) const { return idx % KEY_TYPES_COUNT == 3; }
+    bool IdIsXprv(uint8_t idx) const { return idx % KEY_TYPES_COUNT == 4; }
+
+    //! When initializing the target, populate the list of keys.
+    void Init() {
+        // The data to use as a private key or a seed for an xprv.
+        uint8_t key_data[32] = {1};
+        // Generate keys of all kinds and store them in the keys array.
+        for (uint8_t i{0}; i < TOTAL_KEYS_GENERATED; i++) {
+            key_data[31] = i;
+
+            // If this is a "raw" key, generate a normal privkey. Otherwise generate
+            // an extended one.
+            if (IdIsCPubKey(i) || IdIsXOnlyPubKey(i) || IdIsConstPrivKey(i)) {
+                CKey privkey;
+                privkey.Set(key_data, key_data + 32, true);
```

Comment:
In 845810d94df83747634a60ff59635a670bba1124 "fuzz: add a new, more efficient, descriptor parsing target"

Perhaps also include uncompressed keys?

Reply:
Good call, done.